src/main/java/de/tum/cit/aet/artemis/iris/web/IrisMessageResource.java

```diff
@@ -79,25 +80,34 @@ public ResponseEntity<List<IrisMessage>> getMessages(@PathVariable Long sessionI
 
     /**
      * POST sessions/{sessionId}/messages: Send a new message from the user to the LLM
+     * <p>
+     * Backward compatible: uncommittedFiles is optional and can be omitted.
+     * Legacy requests without uncommittedFiles will work as before.
      *
-     * @param sessionId of the session
-     * @param message   to send
+     * @param sessionId  of the session
+     * @param requestDTO containing message content and optional uncommitted files
      * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the created message, or with status
      *         {@code 404 (Not Found)} if the session could not be found.
      */
     @PostMapping("sessions/{sessionId}/messages")
     @EnforceAtLeastStudent
-    public ResponseEntity<IrisMessage> createMessage(@PathVariable Long sessionId, @RequestBody IrisMessage message) throws URISyntaxException {
+    public ResponseEntity<IrisMessage> createMessage(@PathVariable Long sessionId, @RequestBody IrisMessageRequestDTO requestDTO) throws URISyntaxException {
         var session = irisSessionRepository.findByIdElseThrow(sessionId);
         irisSessionService.checkIsIrisActivated(session);
         var user = userRepository.getUser();
         irisSessionService.checkHasAccessToIrisSession(session, user);
         irisSessionService.checkRateLimit(session, user);
 
+        // Build IrisMessage from DTO
+        var message = new IrisMessage();
+        message.setContent(requestDTO.content());
+        message.setMessageDifferentiator(requestDTO.messageDifferentiator());
```

Comment:
To me the comment does not add value

```suggestion
        var message = new IrisMessage();
        message.setContent(requestDTO.content());
        message.setMessageDifferentiator(requestDTO.messageDifferentiator());
```